### DISCUSSION TITLE
Setting up a webRTCstar transport

### DISCUSSION THREAD

**Discussion by Flexusaurs** [libp2p/js-libp2p#2651] [Q&A] [answered]
Hi !

i've been trying to get a libp2p typescript project going on node 20, but I can't set up proper webRTCstar transports

```
node - v 
Welcome to Node.js v20.11.0.
```

```
const libp2p = await createLibp2p({
        addresses: {
            listen: ['/dns4/wrtc-star1.par.dwebops.pub/tcp/443/wss/p2p-webrtc-star']
          },
          transports: [
           wrtcSTAR.transport
          ],
          peerDiscovery:[
            wrtcSTAR.discovery
          ],
          services: {
          ......
          }
    });
```

for both discovery and transport i have the following errors highlighted:
```

Type '(components: WebRTCStarComponents) => Transport' is not assignable to type '(components: Components) => Transport<ProgressEvent<any, unknown>>'.
  Types of parameters 'components' and 'components' are incompatible.
    Type 'Components' is not assignable to type 'WebRTCStarComponents'.
      Types of property 'peerId' are incompatible.
        Type 'import("/.../node_modules/@libp2p/interface/dist/src/peer-id/index").PeerId' is not assignable to type 'import("/....../node_modules/@libp2p/interface-peer-id/dist/src/index").PeerId'.
          Type 'PeerId' is not assignable to type 'Secp256k1PeerId'.
            Types of property 'type' are incompatible.
 Type 'string' is not assignable to type '"secp256k1"

```

do i have anything missing in my setup? 

thanks in advance

**Comment by maschad** [comment, marked as answer]
[webRTCStar](https://github.com/libp2p/js-libp2p-webrtc-star) has been deprecated for some time now in favour of our [webRTC transport](https://github.com/libp2p/js-libp2p/tree/main/packages/transport-webrtc) 

There's an [example](https://github.com/libp2p/js-libp2p/tree/main/packages/transport-webrtc#example---webrtc) in the README that you can follow for basic setup. If you'd like a more detailed guide of getting this setup feel free to check [this one](https://docs.libp2p.io/guides/getting-started/webrtc/) out. Hope that points you in the right direction.

**Comment by achingbrain** [comment]
Closing as this has been answered - `WebRTCStar` is not supported any more, the modern `@libp2p/webrtc` transport is what needs to be used instead.
